### PR TITLE
feat: wishlist functionality and UI

### PR DESCRIPTION
**Pull request by harshitadaswani**
- added wishlist UI
- added wishlist functionality
- added respective route to it

### CODE REVIEW

File: src/components/cart/Card.jsx [hunk 1]
```diff
@@ -34,7 +37,10 @@ export const Card = () => {
                             <button className="bid-btn br-s p-s m-xs fw-semibold txt-s" onClick={() => cartDispatch({ type: REMOVE_FROM_CART, payload: product })}>
                                 Remove from Cart
                             </button>
-                            <button className="bid-btn br-s p-s m-xs fw-semibold txt-s">
+                            <button className="bid-btn br-s p-s m-xs fw-semibold txt-s" onClick={() => {
```
Comment (ashishnnnnn) [inline src/components/cart/Card.jsx:40]: Liked the usages of utility functions ..

File: src/routes/AppRoutes.jsx [hunk 2]
```diff
@@ -11,6 +11,8 @@ function AppRouter() {
                 <Route path="/login" element={<Login />} />
                 <Route path="/signup" element={<Signup />} />
                 <Route path="/cart" element={<Cart />} />
+                <Route path="/wishlist" element={<Wishlist />} />
+                <Route path="*" element={<ErrorPage />} />
```
Comment (ashishnnnnn) [inline src/routes/AppRoutes.jsx:15]: Good that you have put the error page also.

File: src/utils/WishlistFunctions.jsx [hunk 3]
```diff
@@ -0,0 +1 @@
+export const isProductInWishlist = (state, product) => state.wishlist.find((item) => item.id === product.id);
```
Comment (ashishnnnnn) [inline src/utils/WishlistFunctions.jsx:1]: Liked the use of utility functions to perform checks on wishlist.

**Review by ashishnnnnn — APPROVED** [review]

File: src/components/product/Product.jsx [hunk 4]
```diff
@@ -18,7 +21,14 @@ const Product = () => {
                 return <div className="card p-xs m-xs br-s" key={_id}>
                     <div className="card-img relative">
                         <img src={productImage} alt={title} className="br-s img-responsive" />
-                        <AiFillHeart className="wishlist-btn br-full" size={35} />
+                        <AiFillHeart className="wishlist-btn br-full" size={35} color={wishlistState.wishlist.includes(product) ? "red" : "black"} onClick={() =>
+                            wishlistState.wishlist.includes(product) ? wishlistDispatch({
```
Comment (JantuDeb) [inline src/components/product/Product.jsx:25]: compare id field when you do such comparison

**Review by JantuDeb — APPROVED** [review]
This is well structured code, It is good that you have added 404 page.

File: src/components/product/Product.jsx (hunk 4 above)
Comment (harshitadaswani) [reply src/components/product/Product.jsx:25]: okay will change

**Pull request merged**